third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -61,6 +61,24 @@ findEarlyInsertionPoint(Block *block, Operation *move) {
   return ipnt;
 }
 
+// Check if the operation inLoopOp is inside any scf::ForOp and outLoopOp is not
+// inside the same loop.
+bool checkLoopCondition(mlir::Operation *inLoopOp, mlir::Operation *outLoopOp) {
```

Comment:
I feel naming it as `isCrossLoopBoundary` might be clearer than the current generic name?